File: codeforces/new_year_transpo_SOLVED/new_year.cpp
#include<iostream>
using namespace std;

void expand(pair<int,int> stateTuple, int goal, int* a)
{
    int nextStop = stateTuple.first + stateTuple.second;
    if (nextStop > goal)
    {
        cout << "NO"; return;
    }
    else if (nextStop < goal)
    {
        expand(make_pair(nextStop, a[nextStop]), goal, a);
    }
    else
    {
        cout << "YES";
        return;
    }
    
}

int main()
{
    int n; cin >> n;
    int t; cin >> t;

    int* a = (int*)malloc(n*sizeof(int));
    for (int i = 0; i < n - 1; i ++)
    {
        cin >> a[i];
    }

    expand(make_pair(0, a[0]), t - 1, a);
}

File: codeforces/new_year_transpo_SOLVED/new_year.py
input_list = input()
n, t = int(input_list[0]), int(input_list.split(" ")[1])
goal = t - 1
a = input().split(" ")
aList = []
for i in range(n - 1):
    aList.append(int(a[i]))

def expand(stateTuple):
    nextStop = stateTuple[0] + stateTuple[1]
    if nextStop > goal:
        print("NO")
        return
    elif nextStop < goal:
        expand((nextStop, aList[nextStop]))
    else:
        print("YES")
        return

expand((0,aList[0]))
    